NixOS 20.09: evdi-v1.7.2 fails to build against linuxPackages_5_11

**Describe the bug**
`evdi` kernel module fails to build against `linuxPackages_5_11`.

**To Reproduce**
Steps to reproduce the behavior:
1. Add something that depends on `evdi` (e.g. `displaylink`, add `"displaylink"` to `services.xserver.videoDrivers` list)
2. Set `boot.kernelPackages = lib.mkForce pkgs.linuxPackages_5_11;`
3. Call `sudo nixos-rebuild switch`

**Expected behavior**
`nixos-rebuild` successfully rebuilds and kernel modules that are depending on `evdi` are working as expected.

**Additional context**
It fails to build with `linuxPackages_5_11` (which refers to `5.11.13` kernel version at the moment, and `linuxPackages_latest` is pointing to this version) but successfully builds with `linuxPackages_5_10` for me.

Here are logs chunks that have the information about the failure:

```
/build/source/module/evdi_drv.c:96:3: error: 'struct drm_driver' has no member named 'gem_free_object_unlocked'
   96 |  .gem_free_object_unlocked = evdi_gem_free_object,
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/build/source/module/evdi_drv.c:96:30: error: initialization of 'void (*)(struct drm_device *)' from incompatible pointer type 'void (*)(struct drm_gem_object *)' [-Werror=incompatible-pointer-types]
   96 |  .gem_free_object_unlocked = evdi_gem_free_object,
      |                              ^~~~~~~~~~~~~~~~~~~~
/build/source/module/evdi_drv.c:96:30: note: (near initialization for 'driver.lastclose')
/build/source/module/evdi_drv.c:100:3: error: 'struct drm_driver' has no member named 'gem_vm_ops'
  100 |  .gem_vm_ops = &evdi_gem_vm_ops,
      |   ^~~~~~~~~~
/build/source/module/evdi_drv.c:100:16: error: initialization of 'void (*)(struct drm_device *)' from incompatible pointer type 'const struct vm_operations_struct *' [-Werror=incompatible-pointer-types]
  100 |  .gem_vm_ops = &evdi_gem_vm_ops,
      |                ^
/build/source/module/evdi_drv.c:100:16: note: (near initialization for 'driver.unload')
/build/source/module/evdi_drv.c:100:16: error: initialized field overwritten [-Werror=override-init]
/build/source/module/evdi_drv.c:100:16: note: (near initialization for 'driver.unload')
/build/source/module/evdi_drv.c:114:3: error: 'struct drm_driver' has no member named 'gem_prime_export'; did you mean 'gem_prime_import'?
  114 |  .gem_prime_export = drm_gem_prime_export,
      |   ^~~~~~~~~~~~~~~~
      |   gem_prime_import
/build/source/module/evdi_drv.c:114:22: error: initialization of 'int (*)(struct drm_device *, struct drm_file *, int,  uint32_t *)' {aka 'int (*)(struct drm_device *, struct drm_file *, int,  unsigned int *)'} from incompatible pointer type 'struct dma_buf * (*)(struct drm_gem_object *, int)' [-Werror=incompatible-pointer-types]
  114 |  .gem_prime_export = drm_gem_prime_export,
      |                      ^~~~~~~~~~~~~~~~~~~~
/build/source/module/evdi_drv.c:114:22: note: (near initialization for 'driver.prime_fd_to_handle')
/build/source/module/evdi_drv.c:114:22: error: initialized field overwritten [-Werror=override-init]
/build/source/module/evdi_drv.c:114:22: note: (near initialization for 'driver.prime_fd_to_handle')
/build/source/module/evdi_drv.c:115:3: error: 'struct drm_driver' has no member named 'gem_prime_get_sg_table'; did you mean 'gem_prime_import_sg_table'?
  115 |  .gem_prime_get_sg_table = evdi_prime_get_sg_table,
      |   ^~~~~~~~~~~~~~~~~~~~~~
      |   gem_prime_import_sg_table
/build/source/module/evdi_drv.c:115:28: error: initialization of 'struct drm_gem_object * (*)(struct drm_device *, struct dma_buf *)' from incompatible pointer type 'struct sg_table * (*)(struct drm_gem_object *)' [-Werror=incompatible-pointer-types]
  115 |  .gem_prime_get_sg_table = evdi_prime_get_sg_table,
```

```
/build/source/module/evdi_drv.c:115:28: note: (near initialization for 'driver.gem_prime_import')
/build/source/module/evdi_drv.c:115:28: error: initialized field overwritten [-Werror=override-init]
/build/source/module/evdi_drv.c:115:28: note: (near initialization for 'driver.gem_prime_import')
cc1: all warnings being treated as errors
make[4]: *** [/nix/store/bqr9q4zsi01wj2h4qk0j3nbv8fkd4pbg-linux-5.11.13-dev/lib/modules/5.11.13/source/scripts/Makefile.build:279: /build/source/module/evdi_drv.o] Error 1
make[3]: *** [/nix/store/bqr9q4zsi01wj2h4qk0j3nbv8fkd4pbg-linux-5.11.13-dev/lib/modules/5.11.13/source/Makefile:1820: /build/source/module] Error 2
make[2]: *** [/nix/store/bqr9q4zsi01wj2h4qk0j3nbv8fkd4pbg-linux-5.11.13-dev/lib/modules/5.11.13/source/Makefile:185: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/bqr9q4zsi01wj2h4qk0j3nbv8fkd4pbg-linux-5.11.13-dev/lib/modules/5.11.13/build'
make[1]: *** [Makefile:77: module] Error 2
make[1]: Leaving directory '/build/source/module'
make: *** [Makefile:8: all] Error 2
```

```
builder for '/nix/store/jp6ldzag8vpxx6zl5s4j3icrhlcilvm4-evdi-v1.7.2.drv' failed with exit code 2
```

```
cannot build derivation '/nix/store/rsc3xf6zm5g635z6binjq1ls90hx03z3-kernel-modules.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r5r3fj6j3sc91cgljrb4w7hxynkrj88r-linux-5.11.13-modules.drv': 1 dependencies couldn't be built
```

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
- @thoughtpolice (`linuxPackages_5_11.kernel`)
- @eyJhb (`linuxPackages_5_11.evdi`)
- @nshalman @abbradar @peterhoeg @eyJhb (`displaylink`)

**Metadata**
Please note that I’m using `linuxPackages_5_10` because I couldn’t build it with `linuxPackages_5_11`.

``` sh
nix-shell -p nix-info --run "nix-info -m"
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.29, NixOS, 20.09.3834.c7e905b6a97 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos, nixos-unstable"`
 - channels(wenzel): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

The channel pin is: https://releases.nixos.org/nixos/20.09/nixos-20.09.3834.c7e905b6a97

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- linuxPackages_5_11.kernel
- linuxPackages_5_11.evdi
- displaylink
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/os-specific/linux/displaylink/default.nix
+++ b/pkgs/os-specific/linux/displaylink/default.nix
@@ -10,17 +10,17 @@ let
 
 in stdenv.mkDerivation rec {
   pname = "displaylink";
-  version = "5.3.1.34";
+  version = "5.4.0-55.153";
 
   src = requireFile rec {
     name = "displaylink.zip";
-    sha256 = "1c1kbjgpb71f73qnyl44rvwi6l4ivddq789rwvvh0ahw2jm324hy";
+    sha256 = "1m2l3bnlfwfp94w7khr05npsbysg9mcyi7hi85n78xkd0xdcxml8";
     message = ''
       In order to install the DisplayLink drivers, you must first
       comply with DisplayLink's EULA and download the binaries and
       sources from here:
 
-      https://www.displaylink.com/downloads/file?id=1576
+      https://www.synaptics.com/node/3751
 
       Once you have downloaded the file, please use the following
       commands and re-run the installation:

--- a/pkgs/os-specific/linux/evdi/default.nix
+++ b/pkgs/os-specific/linux/evdi/default.nix
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "evdi";
-  version = "v1.7.2";
+  version = "unstable-20210401";
 
   src = fetchFromGitHub {
     owner = "DisplayLink";
     repo = pname;
-    rev = version;
-    sha256 = "074j0xh037n8mc4isihfz9lap57wvxaxib32pvy6jhjl3wyik632";
+    rev = "b0b3d131b26df62664ca33775679eea7b70c47b1";
+    sha256 = "09apbvdc78bbqzja9z3b1wrwmqkv3k7cn3lll5gsskxjnqbhxk9y";
   };
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
@@ -33,6 +33,6 @@ stdenv.mkDerivation rec {
     platforms = platforms.linux;
     license = with licenses; [ lgpl21 gpl2 ];
     homepage = "https://www.displaylink.com/";
-    broken = with kernel; kernelOlder "5" || stdenv.isAarch64;
+    broken = versionOlder kernel.version "4.19" || stdenv.isAarch64;
   };
 }

